File: src/프로그래머스_Level1/신규 아이디 추천.js
function solution(new_id) {
    new_id = new_id.toLowerCase();
    new_id = new_id.replace(/[^\w\-\.]/g, "");
    new_id = new_id.replace(/[\.]{2,}/g, ".");
    new_id = new_id.replace(/^\./g, "").replace(/\.$/g, "");

    if (!new_id) new_id = "a";

    if (new_id.length >= 16) new_id = new_id.substring(0, 15);
    new_id = new_id.replace(/\.$/g, "");

    if (new_id.length <= 2) new_id = new_id.padEnd(3, new_id[new_id.length - 1]);

    return new_id;
}

solution("...!@BaT#*..y.abcdefghijklm");
solution("z-+.^.");
solution("=.=");
solution("123_.def");
solution("abcdefghijklmn.p");

File: src/프로그래머스_Level1/신규 아이디 추천.py
import re


def solution(new_id):
    new_id = new_id.lower()
    new_id = re.sub("[^\w\-\.]", "", new_id)
    new_id = re.sub("\.{2,}", ".", new_id)
    new_id = re.sub("^\.|\.$", "", new_id)
    new_id = "a" if len(new_id) == 0 else re.sub("\.$", "", new_id[:15])
    new_id = new_id if len(new_id) > 2 else new_id + "".join([new_id[-1] for i in range(3 - len(new_id))])

    return new_id


print(solution("...!@BaT#*..y.abcdefghijklm"))
print(solution("z-+.^."))
print(solution("=.="))
print(solution("123_.def"))
print(solution("abcdefghijklmn.p"))
